### PR TITLE
[SPARK-45260][PYTHON][DOCS] Refine docstring of `count_distinct`

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to improve the docstring of `count_distinct`.

### Why are the changes needed?

For end users, and better usability of PySpark.

### Does this PR introduce _any_ user-facing change?

Yes, it fixes the user facing documentation.

### How was this patch tested?

Manually tested.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -4626,26 +4626,38 @@ def count_distinct(col: "ColumnOrName", *cols: "ColumnOrName") -> Column:
...
+    |count(DISTINCT value)|
+    +---------------------+
+    |                    2|
+    +---------------------+
+
+    Example 2: Counting distinct values of multiple columns
+
+    >>> from pyspark.sql import functions as sf
+    >>> df1 = spark.createDataFrame([(1, 1), (1, 2)], ["value1", "value2"])
+    >>> df1.select(sf.count_distinct(df1.value1, df1.value2)).show()
```
Comment (allisonwang-db) [inline python/pyspark/sql/functions.py:4644]:
```suggestion
    >>> df = spark.createDataFrame([(1, 1), (1, 2)], ["value1", "value2"])
    >>> df.select(sf.count_distinct(df.value1, df.value2)).show()
```

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -4626,26 +4626,38 @@ def count_distinct(col: "ColumnOrName", *cols: "ColumnOrName") -> Column:
...
+    |count(DISTINCT value1, value2)|
+    +------------------------------+
+    |                             2|
+    +------------------------------+
+
+    Example 3: Counting distinct values with column names as strings
+
+    >>> from pyspark.sql import functions as sf
+    >>> df3 = spark.createDataFrame([(1, 1), (1, 2)], ["value1", "value2"])
+    >>> df3.select(sf.count_distinct("value1", "value2")).show()
```
Comment (allisonwang-db) [inline python/pyspark/sql/functions.py:4655]:
```suggestion
    >>> df = spark.createDataFrame([(1, 1), (1, 2)], ["value1", "value2"])
    >>> df.select(sf.count_distinct("value1", "value2")).show()
```

**Review by allisonwang-db — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**